Added new touch action support.

npradeepkb opened the pull request:
test added for only tap touch action. Please review.

Need to add element id as the first argument to all touch actions, 
Touch action methods will accept either element-id (el) , or x and y coordinates (x, y) or all (el, x, y) 

x and y coordinates (x, y) this is completed.

gempesaw commented on lib/Appium.pm:
```diff
@@ -194,7 +194,7 @@ has 'touch_actions' => (
     lazy => 1,
     init_arg => undef,
-    handles => [ qw/tap/ ],
+    handles => [ qw/tap swipe wait moveTo release press longPress/ ],
```
prefer underscore_case for compound words instead of camelCase to match the convention in the rest of the repository

gempesaw commented on lib/Appium/TouchActions.pm:
```diff
@@ -23,24 +23,124 @@ the screen. Values greater than 1 will be interpreted as pixels. (10,
+    };
+
+
+    #$self->execute_script('mobile: tap', $params);
```
no need to keep this in a comment, especially since it's officially deprecated

gempesaw commented on t/TouchActions.t:
```diff
@@ -31,11 +32,11 @@ my $ta = Appium::TouchActions->new(
-    is($actions->{'mobile: tap'}->{y}, $y, 'with the correct y coords');
+    ok(exists $actions->{'action'}->{'tap'}, 'we perform precise taps');
+    is($actions->{'options'}->{x}, $x, 'with the correct x coords');
+    is($actions->{'options'}->{y}, $y, 'with the correct y coords');
```
Are you planning to add tests that hit the rest of the subroutines that you added to TouchActions.pm?

gempesaw commented on lib/Appium/TouchActions.pm:
```diff
@@ -23,24 +23,124 @@ the screen. Values greater than 1 will be interpreted as pixels. (10,
+	my ($self, @coords) = @_;
+
+
+    my $coordinates = {
```
This should probably be called `$options` instead of `$coordinates` since it has `duration` as a key in it, so it's not strictly coordinates anymore.

gempesaw commented on lib/Appium/Commands.pm:
```diff
@@ -34,11 +34,11 @@ has 'get_cmds' => (
+                method => 'POST',
+                url => 'session/:sessionId/touch/perform',
+                no_content_success => 1
+            },
```
oh! nice find, I didn't realize this was already here.

gempesaw commented:
Hey Pradeep, thanks for sending in a PR! Looks like these changes have some outstanding issues right now:

- Tests are failing on Travis for [Commands.t](https://travis-ci.org/appium/perl-client/jobs/151415769#L425) and [TouchActions.t](https://travis-ci.org/appium/perl-client/jobs/151415769#L430), do you need help looking into these?
- Do you plan to add tests for the rest of the new TouchAction subroutines?
- Do you plan to add documentation for the new TouchAction subroutines?
- There are some trailing whitespace issues, but I don't mind cleaning those up on my own during the merge.

Again, thanks so much, looking forward to resolving this one. Cheers!

npradeepkb commented:
Find answers inline

On Mon, Aug 22, 2016 at 11:06 PM, Daniel Gempesaw <notifications@github.com>
wrote:

> Hey Pradeep, thanks for sending in a PR! Looks like these changes have
> some outstanding issues right now:
>
>    - Tests are failing on Travis for Commands.t
>    <https://travis-ci.org/appium/perl-client/jobs/151415769#L425> and
>    TouchActions.t
>    <https://travis-ci.org/appium/perl-client/jobs/151415769#L430>, do you
>    need help looking into these?
>
> *Yes i need some help on this. How can i run this in local server.*

>
>    - Do you plan to add tests for the rest of the new TouchAction
>    subroutines?
>
> *i have added for missing ones, will checkin. *

>
>    - Do you plan to add documentation for the new TouchAction subroutines?
>
> * i have added for missing ones, will checkin.*

>
>    - There are some trailing whitespace issues, but I don't mind cleaning
>    those up on my own during the merge.
>
> *Ok no issue. *

>
>
> Again, thanks so much, looking forward to resolving this one. Cheers!
>

 —

> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/appium/perl-client/pull/13#issuecomment-241489499>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AINJQW2FkXt4fJji1ll3vVWQ4tlbQkYZks5qid4sgaJpZM4Jhz9t>
> .
>

npradeepkb commented:
can you check and accept pull request

gempesaw commented on lib/Appium/TouchActions.pm:
```diff
@@ -6,7 +6,7 @@ use Moo;
     is => 'ro',
     required => 1,
-    handles => [ qw/execute_script/ ]
+    handles => [ qw/prompt/ ]
```
Sorry, I don't understand what this `handles` entry is for. I searched for `prompt` in TouchActions.pm and Appium.pm but didn't find any entries. Do you remember why it's here?

npradeepkb replied:
it should be

*has 'driver' => (*
*    is => 'ro',*
*    required => 1,*
*    handles => [ qw/perform execute_script/ ]*
*);*


*if you want i will fix this.*

On Thu, Sep 8, 2016 at 8:26 PM, Daniel Gempesaw <notifications@github.com>
wrote:

> In lib/Appium/TouchActions.pm
> <https://github.com/appium/perl-client/pull/13#discussion_r78023110>:
>
> > @@ -6,7 +6,7 @@ use Moo;
> >  has 'driver' => (
> >      is => 'ro',
> >      required => 1,
> > -    handles => [ qw/execute_script/ ]
> > +    handles => [ qw/prompt/ ]
>
> Sorry, I don't understand what this handles entry is for. I searched for
> prompt in TouchActions.pm and Appium.pm but didn't find any entries. Do
> you remember why it's here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/appium/perl-client/pull/13/files/35beb40d930706e5686fcc9df8b6dc6ce5917da1#r78023110>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AINJQZi0hDEAby7bkmSXe7NFB8IEBBJGks5qoCI6gaJpZM4Jhz9t>
> .
>

gempesaw commented:
sure, I'm taking a look today.

gempesaw commented:
@npradeepkb thanks for all your hard work on this. I'm doing a few refactors on a different branch to make the code more similar to the [python](https://github.com/appium/python-client/blob/master/appium/webdriver/common/touch_action.py) and [ruby](https://github.com/appium/ruby_lib/blob/master/lib/appium_lib/device/touch_actions.rb) implementations, and I'm also going to expand & clean up the documentation, since there were a few copy/paste errors. I hope to have it merged into master soon. Cheers!

npradeepkb commented:
Thanks.

On Mon, Sep 12, 2016, 7:49 PM Daniel Gempesaw <notifications@github.com>
wrote:

> <https://github.com/npradeepkb>

> @npradeepkb <https://github.com/npradeepkb> thanks for all your hard work
on this. I'm doing a few refactors on a different branch to make the code
more similar to the python
<https://github.com/appium/python-client/blob/master/appium/webdriver/common/touch_action.py>
and ruby
<https://github.com/appium/ruby_lib/blob/master/lib/appium_lib/device/touch_actions.rb>
implementations, and I'm also going to expand & clean up the documentation,
since there were a few copy/paste errors. I hope to have it merged into
master soon. Cheers!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
<https://github.com/appium/perl-client/pull/13#issuecomment-246360144>, or
mute the thread
<https://github.com/notifications/unsubscribe-auth/AINJQXiSehOpqbVV-LFxceFK8MTsWQAKks5qpV4HgaJpZM4Jhz9t>
.
